crates/ruff_server/src/trace.rs

```diff
@@ -0,0 +1,160 @@
+use lsp_types::{notification::Notification, TraceValue};
+use serde::Deserialize;
+use std::sync::{Mutex, OnceLock};
+use tracing::{level_filters::LevelFilter, Event};
+use tracing_subscriber::{layer::SubscriberExt, Layer};
+
+use crate::server::ClientSender;
+
+static LOGGING_SENDER: OnceLock<ClientSender> = OnceLock::new();
+
+static TRACE_VALUE: Mutex<lsp_types::TraceValue> = Mutex::new(lsp_types::TraceValue::Messages);
```

Comment:
The reason I went with `Messages` instead of `Off` as a default is so that `info` events are logged by default if the trace value is unset by the LSP client.